core/embed/trezorhal/optiga/optiga.c

```diff
@@ -634,6 +733,100 @@ int optiga_pin_verify(OPTIGA_UI_PROGRESS ui_progress,
   return OPTIGA_SUCCESS;
 }
 
+int optiga_pin_verify(OPTIGA_UI_PROGRESS ui_progress,
+                      uint8_t stretched_pin[OPTIGA_PIN_SECRET_SIZE]) {
+  // Stretch the PIN more with stretching secrets from the Optiga.
+  optiga_result res = optiga_pin_stretch_secret(ui_progress, stretched_pin);
+  if (res != OPTIGA_SUCCESS) {
+    return res;
+  }
+
+  // Process the stretched PIN using a one-way function before sending it to the
+  // Optiga.
+  uint8_t digest[OPTIGA_PIN_SECRET_SIZE] = {0};
+  hmac_sha256(stretched_pin, OPTIGA_PIN_SECRET_SIZE, NULL, 0, digest);
+
+  // HMAC the digest with the key in OID_PIN_HMAC.
+  uint8_t hmac_buffer[ENCRYPT_SYM_PREFIX_SIZE + OPTIGA_PIN_SECRET_SIZE] = {0};
+  size_t size = 0;
+  res = optiga_encrypt_sym(OPTIGA_SYM_MODE_HMAC_SHA256, OID_PIN_HMAC, digest,
+                           sizeof(digest), hmac_buffer, sizeof(hmac_buffer),
+                           &size);
+  if (res != OPTIGA_SUCCESS) {
+    memzero(digest, sizeof(digest));
```

Comment:
I would move `memzero(digest, sizeof(digest));` just before the `if`.

Reply:
ff74858d83e5adbfa585fa338560317433739651, but it makes no difference because the value is overwritten a few lines later anyway.